flask_unchained/bundles/oauth/views/oauth_controller.py

```diff
@@ -0,0 +1,34 @@
+from flask_unchained import Controller, current_app, request, route, session, url_for, injectable
+
+from ..extensions import OAuth
+
+
+class OAuthController(Controller):
+    oauth: OAuth = injectable
+
+    @route(methods=['GET', 'POST'])
+    def login(self):
+        return self.oauth.github.authorize(callback=url_for(
+            'o_auth_controller.authorized', _external=True, _scheme='https'))
+
+    @route(methods=['GET', 'POST'])
+    def logout(self):
+        session.pop('oauth_token', None)
+        return self.redirect('SECURITY_POST_LOGOUT_REDIRECT_ENDPOINT')
+
+    @route(methods=['GET', 'POST'])
+    def authorized(self):
+        resp = self.oauth.github.authorized_response()
+        if resp is None or resp.get('access_token') is None:
+            return 'Access denied: reason=%s error=%s resp=%s' % (
+                request.args['error'],
+                request.args['error_description'],
+                resp
+            )
+        session['oauth_token'] = (resp['access_token'], '')
+        user = self.oauth.github.get('user')
+
+        user_login = user.data['login']
+        user_email = user.data['email']
```

Comment:
This user stuffs is provider-specific, and I think maybe better suited to a service than the controller.

You might do something like this:

```python
class OAuthController:
    oauth_service: OAuthService = injectable

    @route('/authorized/<string:remote_app>')
    def authorized(self, remote_app):
        # ...
        self.oauth_service.on_authorized(provider)
        return self.redirect()

# flask_unchained/bundles/oauth/services.py
from flask_unchained import BaseService, unchained

@unchained.service('oauth_service')
class OAuthService(BaseService):
    def on_authorized(self, provider):
        pass
```

And then in your app bundle, you'd extend that service to customize the `on_authorized` method (the dependency injection system will make sure that your app bundle's service is the one that gets used, instead of the one from the oauth bundle)